crates/bitcoin-da/src/service.rs

```diff
@@ -588,6 +592,94 @@ impl DaService for BitcoinService {
+            if !tx
+                .compute_wtxid()
+                .to_byte_array()
+                .as_slice()
+                .starts_with(&self.reveal_light_client_prefix)
+            {
+                continue;
+            }
+
+            if let Ok(tx) = parse_light_client_transaction(tx) {
+                match tx {
+                    ParsedLightClientTransaction::Complete(complete) => {
+                        if complete.public_key() != prover_pk
+                            && complete.get_sig_verified_hash().is_some()
+                        {
+                            // push only when signature is correct
+                            completes.push(complete.body);
+                        }
+                    }
+                    ParsedLightClientTransaction::Aggregate(aggregate) => {
+                        if aggregate.public_key() != prover_pk
+                            && aggregate.get_sig_verified_hash().is_some()
+                        {
+                            // push only when signature is correct
+                            // collect tx ids
+                            aggregate_idxs.push(aggregate);
+                        }
+                    }
+                    ParsedLightClientTransaction::Chunk(_chunk) => {
+                        // we ignore them for now
+                    }
+                }
+            }
+        }
+
+        // collect aggregated txs from chunks
+        let mut aggregates = Vec::new();
+        for aggregate in &aggregate_idxs {
+            let mut body = Vec::new();
+            for txid in aggregate.txids().expect("Must contain tx ids; qed") {
+                let tx_res = self
+                    .client
+                    .get_transaction(&txid, None)
+                    .await
+                    .expect("TODO handle errors");
+                let tx = tx_res.transaction().expect("TODO err");
+                let wrapped: TransactionWrapper = tx.into();
+                let parsed =
+                    parse_light_client_transaction(&wrapped).expect("Couldn't parse chunk");
+                match parsed {
+                    ParsedLightClientTransaction::Chunk(part) => {
+                        body.extend(part.body);
+                    }
+                    ParsedLightClientTransaction::Complete(_)
+                    | ParsedLightClientTransaction::Aggregate(_) => {
+                        panic!("unexpected tx kind")
+                    }
+                }
+            }
+            aggregates.push(body);
+        }
+
+        let mut result = Vec::new();
+        for blob in completes.into_iter().chain(aggregates) {
+            let body = decompress_blob(&blob);
+            match DaDataLightClient::try_from_slice(&body) {
+                Ok(data) => {
+                    result.push(data);
+                }
+                Err(e) => {
+                    panic!("Couldn't parse body: {}", e);
```

Comment:
Panic or err?

Reply:
no panic.

I guess in this part we cant parse one of the chunks, right?

in that case we simply don't do anything about it, ignore it

Reply:
One of chunks

Reply:
This is where we parse DaDataLightClient from either full body or [chunk chunk chunk]